PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:8 + add:0, change matches:
-    _inherit = 'account.analytic.line'
-    _inherit = 'hr.employee'
-    _inherit = "hr.leave.type"
-    _inherit = "hr.leave"
-    _inherit = 'project.task'
-    _inherit = 'res.company'
-    _inherit = 'res.config.settings'
-    _inherit = "resource.calendar.leaves"

Total Changes: 27

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/project_timesheet_holidays/models/account_analytic.py
+++ b/addons/project_timesheet_holidays/models/account_analytic.py
@@ -7,7 +7,7 @@ from odoo.osv import expression
 
 
 class AccountAnalyticLine(models.Model):
-    _inherit = 'account.analytic.line'
+    _inherit = ['account.analytic.line']
 
     holiday_id = fields.Many2one("hr.leave", string='Time Off Request', copy=False, index='btree_not_null', export_string_translation=False)
     global_leave_id = fields.Many2one("resource.calendar.leaves", string="Global Time Off", index='btree_not_null', ondelete='cascade', export_string_translation=False)

--- a/addons/project_timesheet_holidays/models/hr_employee.py
+++ b/addons/project_timesheet_holidays/models/hr_employee.py
@@ -5,8 +5,8 @@ from odoo import api, fields, models
 from collections import defaultdict
 
 
-class Employee(models.Model):
-    _inherit = 'hr.employee'
+class HrEmployee(models.Model):
+    _inherit = ['hr.employee']
 
     @api.model_create_multi
     def create(self, vals_list):

--- a/addons/project_timesheet_holidays/models/hr_holidays.py
+++ b/addons/project_timesheet_holidays/models/hr_holidays.py
@@ -5,8 +5,8 @@ from odoo import api, fields, models, _
 from odoo.exceptions import ValidationError
 
 
-class HolidaysType(models.Model):
-    _inherit = "hr.leave.type"
+class HrLeaveType(models.Model):
+    _inherit = ["hr.leave.type"]
 
     timesheet_generate = fields.Boolean(
         'Generate Timesheets', compute='_compute_timesheet_generate', store=True, readonly=False,
@@ -50,8 +50,8 @@ class HolidaysType(models.Model):
                     "leave the internal project and task empty.", holiday_status.name))
 
 
-class Holidays(models.Model):
-    _inherit = "hr.leave"
+class HrLeave(models.Model):
+    _inherit = ["hr.leave"]
 
     timesheet_ids = fields.One2many('account.analytic.line', 'holiday_id', string="Analytic Lines")
 

--- a/addons/project_timesheet_holidays/models/project_task.py
+++ b/addons/project_timesheet_holidays/models/project_task.py
@@ -3,8 +3,9 @@
 
 from odoo import fields, models, _
 
-class Task(models.Model):
-    _inherit = 'project.task'
+
+class ProjectTask(models.Model):
+    _inherit = ['project.task']
 
     leave_types_count = fields.Integer(compute='_compute_leave_types_count', string="Time Off Types Count")
     is_timeoff_task = fields.Boolean("Is Time off Task", compute="_compute_is_timeoff_task", search="_search_is_timeoff_task", export_string_translation=False)

--- a/addons/project_timesheet_holidays/models/res_company.py
+++ b/addons/project_timesheet_holidays/models/res_company.py
@@ -4,8 +4,8 @@
 from odoo import fields, models, _
 
 
-class Company(models.Model):
-    _inherit = 'res.company'
+class ResCompany(models.Model):
+    _inherit = ['res.company']
 
     leave_timesheet_task_id = fields.Many2one(
         'project.task', string="Time Off Task",

--- a/addons/project_timesheet_holidays/models/res_config_settings.py
+++ b/addons/project_timesheet_holidays/models/res_config_settings.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = 'res.config.settings'
+    _inherit = ['res.config.settings']
 
     internal_project_id = fields.Many2one(
         related='company_id.internal_project_id', required=True, string="Internal Project",

--- a/addons/project_timesheet_holidays/models/resource_calendar_leaves.py
+++ b/addons/project_timesheet_holidays/models/resource_calendar_leaves.py
@@ -8,7 +8,7 @@ from odoo import api, fields, models, _
 
 
 class ResourceCalendarLeaves(models.Model):
-    _inherit = "resource.calendar.leaves"
+    _inherit = ["resource.calendar.leaves"]
 
     timesheet_ids = fields.One2many('account.analytic.line', 'global_leave_id', string="Analytic Lines", export_string_translation=False)
 
